app/src/main/java/com/cyberbot/checkers/game/Grid.java

```diff
@@ -369,34 +370,39 @@ int getValue(PlayerNum playerNum, PlayerNum adversaryNum) {
                 } else if (gridEntry.getPlayer() == PlayerNum.SECOND) {
                     value += Math.abs((gridEntry.getY() - getSize()) / 2);
                 }
-            } else if (gridEntry.getPlayer() == adversaryNum) {
-
-                //count pieces
-
-                value -= 10;
-
-                if (gridEntry.getPieceType() == PieceType.KING) {
-                    value -= 30;
-                }
-
-                //prioritize sides
-
-                value -= Math.abs((gridEntry.getX()) + gridEntry.getY() - getSize()) / 4;
+        }
+        return value;
+    }
 
-                //prioritize forward
+    static Grid simulateMove(Grid grid, GridEntry src, Destination dst) {
+        grid.getEntryByCoords(src.getX(), src.getY()).setPlayer(PlayerNum.NOPLAYER);
+        grid.getEntryByCoords(src.getX(), src.getY()).setPieceType(PieceType.UNASSIGNED);
+        for (GridEntry destroyed:dst.getCapturedPieces()){
+            grid.getEntryByCoords(destroyed.getX(),destroyed.getY()).setPieceType(PieceType.UNASSIGNED);
+            grid.getEntryByCoords(destroyed.getX(),destroyed.getY()).setPlayer(PlayerNum.NOPLAYER);
+        }
+        grid.getEntryByCoords(dst.getDestinationEntry().getX(),dst.getDestinationEntry().getY()).setPlayer(PlayerNum.NOPLAYER);
+        grid.getEntryByCoords(dst.getDestinationEntry().getX(),dst.getDestinationEntry().getY()).setPieceType(PieceType.UNASSIGNED);
+        return grid;
+    }
 
-                if (gridEntry.getPlayer() == PlayerNum.FIRST) {
-                    value -= gridEntry.getY() / 2;
-                } else if (gridEntry.getPlayer() == PlayerNum.SECOND) {
-                    value -= Math.abs((gridEntry.getY() - getSize()) / 2);
-                }
+    boolean win(PlayerNum enemy){
+        int enemyNo = 0;
+        for (GridEntry gridEntry : gridEntries){
+            if (gridEntry.getPlayer() == enemy){
+                enemyNo++;
             }
         }
-        return value;
+        return enemyNo == 0;
     }
 
-    static Grid simulateMove(Grid grid, Move move) {
-        grid.attemptMove(move.getStart(), move.getEnd());
-        return grid;
+    boolean loose(PlayerNum player){
```

Comment:
Should probably rename to `didLose` or `lost`, `loose` (not only being a spelling error) would reflect that it makes the player lose.

Reply:
It should also check if the player has moves available. There might still be pieces on the board despite no moves left.

Reply:
renamed to lost and won, move available check added